### ISSUE TITLE
Linux mint 17.1 Cinnamon --> set Guake to current path when opened from within Nemo

### ISSUE THREAD

**Issue by drsood** [Guake/guake#424]
In Linux mint 17.1 Cinnamon, Guake can now be set as the default terminal for Nemo (default file manager) by opening 'Preferred Applications' and choosing Guake as the default terminal.

But the problem is that Guake fails to detect the current path and instead starts with path set as the home folder. How can I configure guake to open up in the current directory and triggered via 'Open in terminal' from within Nemo?

**Comment by cmeeren** [comment]
I'd very much like to see this working correctly. I see that the milestone for this is continually pushed ahead. Are there any real plans to implement this? :)

**Comment by Stibbons** [comment]
I would like to have more information about the use case. I dont really see what exactly fail. Open in current terminal works, but if memo does not change the current directory correctly like bash does, its normal guake fails supporting it.

**Comment by cmeeren** [comment]
Ok, so this should rather be fixed in nemo? By for example adding the possibility of customizing the "Open in terminal" command?

**Comment by cmeeren** [comment]
I've tried making a nemo action for this, placing the following in `~/.local/share/nemo/actions/open_in_guake.nemo_action`:

```
[Nemo Action]
Name=Open in Guake
Comment=Open current folder in Guake
Exec=guake -n %P
Icon-Name=terminal
Selection=none
Extensions=dir
Quote=double
```

This opens a new tab with the correct folder in Guake, but it does not show Guake; I have to do that manually. I've tried `guake -t -n %P`, `guake -n %P && guake`, `guake -n %P && guake -t`, `guake && guake -n %P` (in addition to the above using `;` instead of `&`), but none of them open guake. This is weird, because simply `guake` opens Guake (but of course does not open a new tab with the correct folder).

**Comment by caph1993** [comment]
Try `guake --show -n %P`. It works for me.
I agree it would be nice if we could select the command to run when invoking the default terminal in Nemo

**Comment by CottonEaster** [comment]
' I agree it would be nice if we could select the command to run when invoking the default terminal in Nemo'. So do I, for what it is worth.

**Comment by BeatLink** [comment]
Any updates on this?

**Comment by Davidy22** [comment]
I've done some investigating and I have a branch with some very rough changes that will do this [here](https://github.com/Davidy22/guake/tree/openinterm). This change does have side effects though; I have not yet identified a way to distinguish "Open in terminal" from invoking guake from the start menu or terminal. Current behavior when you run guake from the start menu or terminal when guake is already running is that guake is toggled. If there is no way to distinguish nemo invocation and other methods of opening, the new behavior will be that running guake from anywhere when guake is already running will create a new tab.

I have done some investigation into other terminal programs to see how they behave and basically all of them spawn a new window no matter where you invoke them from. This implies to me that 

1. There might not be a way to distinguish a file manager's open in terminal and
2. This is just the standard way that terminal emulators handle this

Although, one of the emulators I tested did actually have a configurable option to behave the way guake currently does, although its default is still open a new window. I think making this an option is probably how we go about it, and we make the default for the option open a new tab because opening a new window like all the other terminal emulators do is kind of not great for guake specifically. If we didn't make this an option, or even if we did make it an option and just changed the default, this will make it so that opening guake from the menu or terminal while guake is already running will behave differently, opening a new tab, which might be surprising. it would fix guake up to match the expectations of the file browsers though. This behavioral change maybe needs discussion, how many people are opening guake from the menu expecting it to not create a new tab on them? If that's not really an issue, I can polish this up and make a pull request.
